### PR TITLE
port field_effect_helpers from emerald

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: src/field_effect_helpers.c [hunk 1]
```diff
@@ -0,0 +1,1410 @@
...
+    }
+    else
+    {
+        objectEvent = &gObjectEvents[objectEventId];
+        if ((objectEvent->currentCoords.x != sprite->data[1] || objectEvent->currentCoords.y != sprite->data[2]) && (objectEvent->previousCoords.x != sprite->data[1] || objectEvent->previousCoords.y != sprite->data[2]))
+            sprite->data[7] = TRUE;
+
+        metatileBehavior = 0;
+        if (sprite->animCmdIndex == 0)
+            metatileBehavior = 4;
```
Comment (jiangzhengwenjz) [inline src/field_effect_helpers.c]: do constants in `include/constants/metatile_behaviors.h` fit here?

File: src/field_effect_helpers.c [hunk 2]
```diff
@@ -0,0 +1,1410 @@
...
+    if (TryGetObjectEventIdByLocalIdAndMap(gFieldEffectArguments[0], gFieldEffectArguments[1], gFieldEffectArguments[2], &spriteId))
+    {
+        FieldEffectActiveListRemove(fldEff);
+        return MAX_SPRITES;
+    }
+    spriteId = CreateSpriteAtEnd(gFieldEffectObjectTemplatePointers[templateIdx], 0, 0, 0);
+    if (spriteId != MAX_SPRITES)
+    {
+        sprite = &gSprites[spriteId];
+        sprite->coordOffsetEnabled ++;
```
Comment (jiangzhengwenjz) [inline src/field_effect_helpers.c]: no space before ++ (6 instances in this file)

File: src/field_effect_helpers.c (hunk 1 above)
Comment (PikalaxALT) [reply src/field_effect_helpers.c]: The variable might be misnamed

Comment (PikalaxALT) [reply src/field_effect_helpers.c]: Or misused, in this context

**Pull request merged**